atlp-rwanda/atlp-devpulse-fn
# Fix fault alerts

### Summary
This PR resolves fault alerts currently reported by the system.

### Related Issues
Closes #163

### Changes Made
- changed toasts which were not meaningful to user when certain page is loaded
- Fixed the issue of user not being able to get data due to token validity
- Updated error messages for better clarity.

### How to Test
1. login as applicant and check for new toast
2. login as superAdmin 
credentials: email: `admin@example.com` password: `password123`  to check fault alerts resolved on programs page and applications page
3. the data now are now being correctly retrieved and shown and toasts are wrt the current login situation of user.

### Screenshots
N/A

### Additional Notes
This PR also refactors some of the error handling logic for better readability.
